policy-controller/core/src/lib.rs

```diff
@@ -101,10 +102,17 @@ impl Ord for InboundHttpRouteRef {
         match (self, other) {
             (Self::Default(a), Self::Default(b)) => a.cmp(b),
             (Self::Linkerd(a), Self::Linkerd(b)) => a.cmp(b),
-            // Route resources are always preferred over default resources, so they should sort
-            // first in a list.
+            (Self::Probe, Self::Probe) => std::cmp::Ordering::Equal,
```

Comment:
this is a lot of different combinations to keep track of.  what do you think about something like:

```
fn priority(routeRef: &InboundHttpRouteRef) -> u8 {
    match routeRef {
        Self::Linkerd(_) => 1,
        Self::Probe => 2,
        Self::Default(_) => 3,
    }
}

fn cmp(&self, other: &Self) -> std::cmp::Ordering {
    match (self, other) {
           (Self::Default(a), Self::Default(b)) => a.cmp(b),
            (Self::Linkerd(a), Self::Linkerd(b)) => a.cmp(b),
            _ => priority(self).cmp(priority(other)),
    }
}
```

or something along those lines?